sklearn/pipeline.py

```diff
@@ -1005,8 +1009,15 @@ def _iter(self):
         'drop' transformers.
         """
         get_weight = (self.transformer_weights or {}).get
+
         return (
-            (name, trans, get_weight(name))
+            (
```

Comment:
I think there is enough `if` logic here to refactor `_iter` into a generator:

```python
    def _iter(self):
        get_weight = (self.transformer_weights or {}).get

        for name, trans in self.transformer_list:
            if trans == "drop":
                continue
            if trans == "passthrough":
                trans = FunctionTransformer()
            yield (name, trans, get_weight(name))
```

Reply:
Yep this is more readable